source-code/app/src/main/java/org/buildmlearn/toolkit/utilities/RestoreThread.java

```diff
@@ -66,8 +74,26 @@ public void run() {
             }
 
             if (data == null) {
-                if (listener != null)
-                    listener.onFail();
+                if (listener != null){
+                    Handler mHandler=new Handler(Looper.getMainLooper());
```

Comment:
No need of adding it here. Bcoz its already implemented in https://github.com/BuildmLearn/BuildmLearn-Toolkit-Android/blob/master/source-code/app/src/main/java/org/buildmlearn/toolkit/fragment/SettingsFragment.java#L129-L138 through a listener.